mono/metadata/sgen-thread-pool.c

```diff
@@ -39,6 +39,9 @@ static SgenThreadPoolThreadInitFunc thread_init_func;
 static SgenThreadPoolIdleJobFunc idle_job_func;
 static SgenThreadPoolContinueIdleJobFunc continue_idle_job_func;
 
+static int threadpool_shutdown;
+static int thread_finished;
```

Comment:
It seems to me that we always call a wait function between accesses to both of these variables, so I'd expect that the compiler cannot assume they haven't changed. Except in the thread loop when threadpool_shutdown is true, but it should never change from true to false. What do you expect 'volatile' to change here?